composeApp/src/commonMain/kotlin/com/developersbreach/kotlindictionarymultiplatform/ui/screens/detail/DetailContent.kt

```diff
@@ -1,28 +1,83 @@
 package com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail
 
-import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.rememberLazyListState
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
 import com.developersbreach.kotlindictionarymultiplatform.data.detail.model.KotlinTopicDetails
 import com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail.components.IntroductionSection
 import com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail.components.PitfallsSection
 import com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail.components.RelatedTopicsSection
-import com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail.components.SectionsList
+import com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail.components.SectionBlock
 import com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail.components.SyntaxSection
 import com.developersbreach.kotlindictionarymultiplatform.ui.screens.detail.components.TableOfContents
+import kotlindictionarymultiplatform.composeapp.generated.resources.Res
+import kotlindictionarymultiplatform.composeapp.generated.resources.bullet_item
+import kotlindictionarymultiplatform.composeapp.generated.resources.introduction_bullet
+import kotlindictionarymultiplatform.composeapp.generated.resources.pitfalls_bullet
+import kotlindictionarymultiplatform.composeapp.generated.resources.related_topics_bullet
+import kotlindictionarymultiplatform.composeapp.generated.resources.syntax_bullet
+import kotlinx.coroutines.launch
+import org.jetbrains.compose.resources.stringResource
 
 @Composable
 fun DetailContent(
     topic: KotlinTopicDetails,
     modifier: Modifier = Modifier,
 ) {
-    Column(modifier = modifier.fillMaxSize()) {
-        TableOfContents(topic)
-        IntroductionSection(topic)
-        SyntaxSection(topic)
-        SectionsList(topic)
-        PitfallsSection(topic)
-        RelatedTopicsSection(topic)
+    val listState = rememberLazyListState()
+    val coroutineScope = rememberCoroutineScope()
+
+    val tocItems = buildList {
```

Comment:
Building your table of contents inside the Composable means your UI is doing work it shouldn’t.
Composables should focus on rendering, this list building logic belongs in the ViewModel .
Please move it to ViewModel.